### PR TITLE
[RFR] move stateful to peer dependencies and devDependencies within modules, add them to react-admin as dependencies

### PR DESCRIPTION
**Pull request by kopax**
- Fix: #1683

FYI: I have created a public gist with the [list of all known peer dependencies in npm](https://gist.github.com/kopax/c814374c5f34c230049e661d5a942129), feel free to add some.

### CODE REVIEW

**Comment by fzaninotto** [comment]
I've fixed the bug in `next` in b5828cd0b61fcf8d25c9649cef47963576befe70 and relaunched the build of your PR.

**Comment by fzaninotto** [comment]
Now e2e tests seem to fail for a valid reason, please check.

**Comment by kopax** [comment]
> Now e2e tests seem to fail for a valid reason, please check.

Is there a way to prevent e2e test from starting/closing so fast, so I can trace and debug?

**Comment by fzaninotto** [comment]
Just call `make run` and test manually.

File: yarn.lock [hunk 1]
```diff
@@ -781,6 +781,28 @@
     "@babel/helper-plugin-utils" "^7.0.0"
     "@babel/plugin-transform-typescript" "^7.3.2"
 
+"@babel/runtime@7.0.0-beta.42":
```
Comment (fzaninotto) [inline yarn.lock]: Something is wrong, you have a lot of duplicated dependencies.

**Comment by kopax** [comment]
unit and e2e now pass.

File: packages/ra-tree-ui-materialui/package.json [hunk 2]
```diff
@@ -36,23 +36,30 @@
...
+        "react": "^16.9.0",
+        "react-dom": "^16.9.0",
+        "rimraf": "^2.6.3"
+    },
     "peerDependencies": {
+        "final-form": "^4.18.4",
         "ra-core": "^3.0.0-alpha.0",
-        "ra-ui-materialui": "^2.5.0",
+        "ra-tree-core": "^3.0.0-beta.0",
+        "ra-ui-materialui": "^3.0.0-beta.",
```
[the remark this answers is not in the document]
Comment (kopax) [reply packages/ra-tree-ui-materialui/package.json]: sorry about that.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**